marshalers.go

```diff
@@ -92,6 +92,9 @@ var bytesReaderPool = sync.Pool{
 //
 // Prefer using Map.unmarshalKey and Map.unmarshalValue if possible, since
 // those have special cases that allow more types to be encoded.
+//
+// The common int32 and int64 types are directly handled to avoid
```

Comment:
Hey, thanks for reviewing !
I added the comment as it might not necessarily be obvious to readers why we explicitly handle these types when the default case already does it.
I have no issue to remove the comment in case other reviewers agree.

Reply:
Sure, might be useful for posterity. Thanks!